autogen/main/cluster.tf.tmpl

```diff
@@ -54,6 +54,9 @@ resource "google_container_cluster" "primary" {
     }
   }
 {% if beta_cluster %}
+  cost_management_config {
+    enabled = var.enable_cost_allocation
+  }
```

Comment:
Let's make this a dynamic block to prevent issues like https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1255

Reply:
Changes made and Commit Pushed